test/TesterAzureUtils/AzureMembershipTableTests.cs

```diff
@@ -17,22 +19,28 @@ namespace Tester.AzureUtils
     [TestCategory("Membership"), TestCategory("Azure")]
     public class AzureMembershipTableTests : MembershipTableTestsBase, IClassFixture<AzureStorageBasicTests>
     {
-        public AzureMembershipTableTests(ConnectionStringFixture fixture, TestEnvironmentFixture environment) : base(fixture, environment)
+        public AzureMembershipTableTests(ConnectionStringFixture fixture, TestEnvironmentFixture environment) : base(fixture, environment, CreateFilters())
         {
-            LogManager.AddTraceLevelOverride("AzureTableDataManager", Severity.Verbose3);
-            LogManager.AddTraceLevelOverride("OrleansSiloInstanceManager", Severity.Verbose3);
-            LogManager.AddTraceLevelOverride("Storage", Severity.Verbose3);
+        }
+
+        private static LoggerFilterOptions CreateFilters()
+        {
+            var filters = new LoggerFilterOptions();
+            filters.AddFilter("AzureTableDataManager", LogLevel.Trace);
+            filters.AddFilter("OrleansSiloInstanceManager", LogLevel.Trace);
+            filters.AddFilter("Storage", LogLevel.Trace);
+            return filters;
         }
 
         protected override IMembershipTable CreateMembershipTable(Logger logger)
         {
             TestUtils.CheckForAzureStorage();
-            return new AzureBasedMembershipTable();
+            return new AzureBasedMembershipTable(loggerFactory);
         }
 
         protected override IGatewayListProvider CreateGatewayListProvider(Logger logger)
         {
-            return new AzureGatewayListProvider();
+            return new AzureGatewayListProvider(loggerFactory);
```

Comment:
You are being passed in the Logger. Maybe you can rework the base class to pass in ILoggerFactory instead if that's what's needed (but can't see the base class in this PR for some reason, so not sure if this is accurate guidance)

Reply:
not sure I follow, why should I pass in the logger?

Reply:
this loggerFactory is from base class. Since all those tests share similar logging config except the filters. So I only pass in filters in constructor. And base class created the loggerFactory, considering the filters